Gem/Code/Source/RHI/TriangleExampleComponent.cpp

```diff
@@ -214,9 +214,9 @@ namespace AtomSampleViewer
                 drawItem.m_arguments = drawIndexed;
                 drawItem.m_pipelineState = m_pipelineState.get();
                 drawItem.m_indexBufferView = &indexBufferView;
-                drawItem.m_shaderResourceGroupCount = RHI::ArraySize(shaderResourceGroups);
+                drawItem.m_shaderResourceGroupCount = static_cast<uint8_t>(RHI::ArraySize(shaderResourceGroups));
                 drawItem.m_shaderResourceGroups = shaderResourceGroups;
-                drawItem.m_streamBufferViewCount = static_cast<uint32_t>(m_streamBufferViews.size());
+                drawItem.m_streamBufferViewCount = static_cast<uint8_t>(m_streamBufferViews.size());
```

Comment:
Should these changes where the type is narrowed from 32 bits to 8 be using aznumeric_cast instead?

Reply:
I am going with `static_cast` instead because 
1) that is what is closer to what the compiler is doing; 
2) we found that aznumeric_cast usage across the board increases build time by ~5%
3) aznumeric_cast communicates that the casting is intentional, but I want that decision to be made by each team. Teams could choose to do something different. E.g. EMotionFX refactored their types to reduce casting